pkg/dataobj/uploader/metrics.go

```diff
@@ -4,30 +4,38 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+var (
+	labelStatus   = "status"
+	statusSuccess = "success"
+	statusFailure = "failure"
+)
+
 type metrics struct {
-	uploadTime     prometheus.Histogram
-	uploadFailures prometheus.Counter
-	shaPrefixSize  prometheus.Gauge
+	uploadCount   *prometheus.CounterVec
+	uploadTime    prometheus.Histogram
+	uploadSize    *prometheus.HistogramVec
+	shaPrefixSize prometheus.Gauge
 }
 
 func newMetrics(shaPrefixSize int) *metrics {
 	metrics := &metrics{
-		uploadFailures: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "loki_dataobj_consumer_upload_failures_total",
-			Help: "Total number of upload failures",
-		}),
+		uploadCount: prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
A small nit. It's somewhat discouraged to have success and failure counters, because it makes a bunch of calculations more verbose (https://promlabs.com/blog/2023/09/19/errors-successes-totals-which-metrics-should-i-expose-to-prometheus/#not-recommended-exposing-errors-and-successes). Instead, we should have total and failure counters like `loki_dataobj_uploader_uploads_total` and `loki_dataobj_uploader_upload_failures_total`.

Reply:
Personally I cannot fully follow the argument, but happy to change it to two separate metrics, one for failures and one for total